packages/nimble-components/src/table-column/base/models/column-validator.ts

```diff
@@ -0,0 +1,61 @@
+import type { ValidityObject } from '../../../table/types';
+import type { ColumnInternals } from './column-internals';
+
+type ObjectFromList<T extends readonly string[]> = {
+    [K in T extends readonly (infer U)[] ? U : never]: boolean;
+};
+
+/**
+ * Base column validator
+ */
+export class ColumnValidator<ValidityFlagNames extends readonly string[]> {
+    protected configValidity: ObjectFromList<ValidityFlagNames>;
+
+    public constructor(
+        private readonly columnInternals: ColumnInternals<unknown>,
+        configValidityKeys: ValidityFlagNames
+    ) {
+        type ConfigValidity = typeof this.configValidity;
+        this.configValidity = configValidityKeys.reduce(
+            (r, key): ConfigValidity => {
+                return {
+                    ...r,
+                    [key]: false
+                };
+            },
+            // eslint-disable-next-line @typescript-eslint/prefer-reduce-type-parameter
```

Comment:
[The docs for this rule](https://typescript-eslint.io/rules/prefer-reduce-type-parameter/) offer guidance about passing a generic type parameter to the `reduce()` function as a way to fix this violation and remove this `as`. Could you try that? If it doesn't work quickly, the suppression is ok but I'd rather we try to fix them.

Reply:
I had tried that, but something is not quite right with the type checking when I do that. It thinks that `reduce()` returns a `string`:
![image](https://github.com/ni/nimble/assets/7282195/fdaa0dba-615c-4142-898f-4ff44c9adc1f)
It also complains about the type of the initial accumulator object:
![image](https://github.com/ni/nimble/assets/7282195/61ababbf-0213-4ffd-899f-624b8655fa1d)
I wasn't sure how to fix these, so I just went with a suppression.

Reply:
Interesting. I also spent a bit of time trying to understand what's going on. I didn't see anything wrong with what you're doing. I did find this TypeScript issue which may be related: https://github.com/microsoft/TypeScript/issues/46438

But I'm ok with leaving this suppressed.